src/App.tsx

```diff
@@ -9,36 +9,109 @@ import { PostDetails } from './components/PostDetails';
+
+  const getUserPosts = async (user: User) => {
+    setIsError(false);
+    setIsLoadingPosts(true);
+
+    try {
+      const response = await getPosts(user.id);
+
+      setCurrentUserPosts(response);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoadingPosts(false);
+    }
+  };
+
+  const hideCommemts = () => {
+    setActivePost(null);
+    setCurrentUserPosts([]);
+  };
+
+  const handleUserSelect = (user: User) => {
+    setCurrentUser(user);
+  };
+
+  const handleSelectActivePost = (post: Post) => {
+    setActivePost(post);
+  };
+
+  useEffect(() => {
+    getUsers()
+      .then(value => {
+        setUsers(value);
+      });
+  }, []);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  currentUser={currentUser}
+                  onUserSelect={handleUserSelect}
+                  getUserPosts={getUserPosts}
+                  hideCommemts={hideCommemts}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!currentUser && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {isLoadingPosts && (
+                  <Loader />
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {isError && (
+                  <div
+                    className={classNames(
+                      'notification',
+                      { 'is-danger': isError },
+                    )}
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {(currentUserPosts.length < 1
+                  && currentUser
+                  && !isLoadingPosts
+                ) && (
+                  <div className="notification is-warning" data-cy="NoPostsYet">
+                    No posts yet
+                  </div>
+                )}
+
+                {(currentUserPosts.length > 0) && (
+                  <PostsList
+                    currentUserPosts={currentUserPosts}
+                    activePost={activePost}
+                    setActivePost={handleSelectActivePost}
```

Comment:
sorry, probably my previous comment confused you
`handleSelectActivePost` was an example - that it's better to create a separate function (if it's possible) above than pass an anonymous function

but here you can directly pass `setActivePost` - it accepts the same argument
```suggestion
                    setActivePost={setActivePost}
```